eng/automation/GenerateAndBuildLib.ps1

```diff
@@ -0,0 +1,149 @@
+    try
+    {
+        $content = Get-Content .\$AUTOREST_CONFIG_FILE -Raw
+        if ($content -match $swaggerInfoRegex)
+        {
+            return $matches["org"], $matches["specName"], $matches["commitID"]
+        }
+        if ($content -match $rawSwaggerInfoRegex)
+        {
+            return $matches["org"], $matches["specName"], $matches["commitID"]
+        }
+        if ($content -match $swaggerNoCommitRegex)
+        {
+            return $matches["org"], $matches["specName"], ""
+        }
+    }
+    catch
+    {
+        Write-Error "Error parsing swagger info"
+        Write-Error $_
+    }
+    Write-Host "Cannot find swagger info"
+    exit 1
+}
+
+function Get-InputJsonFileInfo()
+{
+    param(
+        [string]$dir,
+        [string]$AUTOREST_CONFIG_FILE = "autorest.md"
+    )
+    Set-Location $dir
+    $inputfileInfoRegex = ".*github.*.com\/(?<org>.*)\/azure-rest-api-specs\/blob\/(?<commitID>[0-9a-f]{40})\/specification\/(?<specName>.*)\/data-plane\/?<filePath>"
+    $rawInputfileInfoRegex = ".*github.*.com\/(?<org>.*)\/azure-rest-api-specs\/(?<commitID>[0-9a-f]{40})\/specification\/(?<specName>.*)\/data-plane\/?<filePath>"
+    $inputfileNoCommitRegex = ".*github.*.com\/(?<org>.*)\/azure-rest-api-specs\/(blob\/)?(?<branch>.*)\/specification\/(?<specName>.*)\/data-plane\/?<filePath>"
+    try
+    {
+        $content = Get-Content .\$AUTOREST_CONFIG_FILE -Raw
+        if ($content -match $inputfileInfoRegex)
+        {
+            return $matches["org"], $matches["specName"], $matches["commitID"], $matches["filePath"]
+        }
+        if ($content -match $rawInputfileInfoRegex)
+        {
+            return $matches["org"], $matches["specName"], $matches["commitID"], $matches["filePath"]
+        }
+        if ($content -match $inputfileNoCommitRegex)
+        {
+            return $matches["org"], $matches["specName"], "", $matches["filePath"]
+        }
+    }
+    catch
+    {
+        Write-Error "Error parsing input file info"
+        Write-Error $_
+    }
+    Write-Host "Cannot find input file info"
+    exit 1
+}
+function New-DataPlanePackageFolder() {
+  param(
+      [string]$service,
+      [string]$namespace,
+      [string]$sdkPath = "",
+      [string]$inputfile = "",
+      [string]$securityScope,
+      [string]$AUTOREST_CONFIG_FILE = "autorest.md",
+      [string]$outputJsonFile = "output.json"
+  )
+
+  $sdkPath = $sdkPath -replace "\\", "/"
+
+  $projectFolder="$sdkPath/sdk/$service/$namespace"
+  if (Test-Path -Path $projectFolder) {
+    Write-Host "Path exists!"
+  } else {
+    Write-Host "Path doesn't exist. create template."
+    if ($inputfile -eq "") {
+        Write-Error "Error: input file should not be empty."
+        exit 1
+    }
+    dotnet new -i $sdkPath/eng/templates/Azure.ServiceTemplate.Template
+    $projectFolder="$sdkPath/sdk/$service/$namespace"
```

Comment:
`$projectFolder` is defined above already.

Reply:
It has been changes. In above, we use the regex to search if there is any existing project folder. If not, create one, and set the projectFolder name correctly.

Reply:
removed